### ISSUE TITLE
Testing ContainerHost implementation - coroutine scope issues

### ISSUE THREAD

**Issue by mz-efento** [orbit-mvi/orbit-mvi#176]
- The problem
 Consider following example of simple `ConteinerHost` implementation. Notice, that `CounterViewModel` receives `coroutineScope` as a parameter, which is then used to create `container()`
 
 ```
 class CounterViewModel(
    coroutineScope: CoroutineScope
) : ContainerHost<CounterState, CounterSideEffect> {
    override val container: Container<CounterState, CounterSideEffect> = coroutineScope.container(
        CounterState(ctr = 0)
    )

    fun incrementCounter() = intent {
        reduce { state.copy(ctr = state.ctr + 1) }
    }
}

data class CounterState(
    val ctr: Int
)

sealed class CounterSideEffect {}
 ```
 
---

The test for this ViewModel may look like this:
```
class CounterViewModelTest {

    @Test
    fun `increment test`() = runTest {
        CounterViewModel(coroutineScope = this).test(testScope = this) {
            expectInitialState()

            invokeIntent { incrementCounter() }

            expectState(CounterState(ctr = 1))
        }
    }
```

But it turns out that this test throws an Exception:
`kotlinx.coroutines.test.UncompletedCoroutinesError: After waiting for 10s, the test coroutine is not completing, there were active child jobs: `

- The (probable) cause -> scopes

After playing around a little bit I noticed that changing the `coroutineScope` parameter of `CounterViewModel` fixes the issue and the test runs just fine.

```
CounterViewModel(coroutineScope = this).test(testScope = this) //Exception
changed to
CounterViewModel(coroutineScope = backgroundScope).test(testScope = this) //OK
```

- The questions
Why is that?
What it considered the best practice for testing such `ContainerHost` implementations and which coroutine scopes should be used?

Thanks!

**Comment by TWiStErRob** [comment]
Oooh, that sounds much better than what I used: `MyContainerHost(this + Job())`, based on the documentation of `backgroundScope`, it's meant for this.

Whatever the answer will be from the maintainers, I would love to see it listed on https://orbit-mvi.org/Test/new in a section called "Testing pure ContainerHost" (without using a ViewModel).

**Comment by Rosomack** [comment]
TL;DR You need to use `this.backgroundScope` for this :)
